hoot-core/src/main/cpp/hoot/core/io/HootApiDb.cpp

```diff
@@ -1319,15 +1345,33 @@ bool HootApiDb::currentUserCanAccessMap(const long mapId, const bool write)
     isPublic = _getMapPermissionsById->value(1).toBool();
     LOG_VART(isPublic);
   }
+  //  Only query for admin permissions if the map isn't owned by the current user
+  if (_currUserId != userId)
+  {
+    //  Create the admin query if it doesn't exist
+    if (!_isUserAdmin)
+    {
+      _isUserAdmin.reset(new QSqlQuery(_db));
+      _isUserAdmin->prepare(
+        "SELECT (u.privileges -> 'admin')::boolean AS is_admin FROM users u WHERE id=:id;");
+    }
+    //  Query for the admin privileges of the current user
+    _isUserAdmin->bindValue(":id", (qlonglong)_currUserId);
+    if (!_isUserAdmin->exec())
+      throw HootException(_isUserAdmin->lastError().text());
+    else if (_isUserAdmin->next())
+      isAdmin = _isUserAdmin->value(0).toBool();
+    _isUserAdmin->finish();
+  }
   _getMapPermissionsById->finish();
 
   if (write)
   {
-    return _currUserId == userId;
+    return _currUserId == userId || isAdmin;
   }
   else
   {
-    return isPublic || _currUserId == userId;
+    return isPublic || _currUserId == userId || isAdmin;
```

Comment:
@brianhatchl this is the section that I want you to look at, essentially `HootApiDb::currentUserCanAccessMap` used to check if the owner of the map is the current user or the folder that the map is in is public.  Here I only check if the `admin=>true` privilege is set when the map owner isn't the current user.